ctf/setup.py

```diff
@@ -49,3 +54,8 @@ def build_challenges():
                     db.session.commit()
                 except IntegrityError:
                     db.session.rollback()
+                    challenge = Challenge.query.filter_by(
+                            title=problem['title'])
+                    problem_dict.update({'id': challenge.first().id})
```

Comment:
This feels very implicit, counting on the problem dict to contain the correct keys for our Challenge ORM object.  Just looking at this function, it's hard to figure out what exactly is supposed to be in this dict.  Do you think it would make more sense for ``build_problem_options`` to give us a  new``Challenge`` object instead of a dict, and just update the ID on that object?